lib/src/visitor/serialize.dart

```diff
@@ -1085,23 +1090,41 @@ class _SerializeVisitor
       return;
     }
 
-    _writeLineFeed();
+    bool indentNextChild;
+    if (_shouldWriteLineFeedBeforeChildren(children)) {
+      _writeLineFeed();
+      indentNextChild = true;
+    } else {
+      _writeOptionalSpace();
+      indentNextChild = false;
+    }
+
     CssNode previous;
-    _indent(() {
-      for (var i = 0; i < children.length; i++) {
-        var child = children[i];
-        if (_isInvisible(child)) continue;
+    for (var i = 0; i < children.length; i++) {
```

Comment:
```suggestion
    for (var child in children) {
```

I don't know why this was using an int-style for loop in the first place.